articles/django-rest-api-mongodb-notes-app/django-rest-api-mongodb-notes-app.md

```diff
@@ -0,0 +1,225 @@
+# Build a Notes App API with Django REST Framework and MongoDB
+
+## Introduction
+
+MongoDB is a NoSQL database system. It stores data in JSON-like format instead of the traditional row-column format. The Django REST framework is a powerful toolkit for building extensible APIs. The Django framework is a Python framework for web development.
+
+You will learn how to use the powerful Django REST framework and MongoDB in your web projects. You will learn how to put in place CRUD features in a notes app.
+
+In this post, you will:
+
+- Understand to integrate MongoDB into Django projects
+- Learn how to set up and run Django REST API
+- Create CRUD features in Django REST API
+- Build a Notes app API
+
+## Prerequisites
+
+- Python 3 installed on your machine. You can install Python 3 from the [Python website](https://www.python.org/download/releases/3.0/).
+- Experience with Python and Django
+
+## Setting up MongoDB Server
+
+You may download and install the MongoDB database server from the [official MongoDB website](https://www.mongodb.com/try/download/community).
+
+## Setting up the Virtual Environment
+
+We will install `virtualenv` to enable a virtual environment for our project. It will enable one to isolate a project and its dependencies from other projects on your machine. Run the following command to install `virtualenv`:
+
+```
+python -m pip install --user virtualenv
+```
```

Comment:
Use bash syntax highlighting for this and similar code blocks. ie.
```markdown
```bash
$ python -m pip install --user virtualenv
```